src/OpenIddict.Core/OpenIddictController.cs

```diff
@@ -86,7 +88,37 @@ public class OpenIddictController<TUser, TApplication> : Controller where TUser
                 });
             }
 
-            return View("Authorize", Tuple.Create(request, await Manager.GetDisplayNameAsync(application)));
+            // Get all scopes associated with the current application and ensure all scopes specified in the authorization request 
+            // belongs to the current application or fail with an error, note that we automatically strip out standard **openid** scopes.
+            var requestedScopes = request.GetCustomScopes();
+
+            // Get all application scopes
+            var applicationScopes = await Manager.GetScopesByApplicationAsync(application);
+
+            // Get all request scopes
+            var requestScopes = await Manager.GetAuthorizationRequesteScopesAsync(requestedScopes);
```

Comment:
To get scopes from db/somewhere for current request. In this point we have AR scopes (list of strings) and Application scopes (lists of ??) since ```where TScope : class``` so we need to demand all scope related work to the Store.